### PR TITLE
Add terradoc

### PR DESCRIPTION
**Pull request by kassianh**
Recreated README with terradoc, checked documentation for errors or inconsistencies.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,10 @@
-[<img src="https://raw.githubusercontent.com/mineiros-io/brand/3bffd30e8bdbbde32c143e2650b2faa55f1df3ea/mineiros-primary-logo.svg" width="400"/>][homepage]
+[<img src="https://raw.githubusercontent.com/mineiros-io/brand/3bffd30e8bdbbde32c143e2650b2faa55f1df3ea/mineiros-primary-logo.svg" width="400"/>](https://mineiros.io/?ref=terraform-aws-cognito-user-pool)
 
-[![Build Status][badge-build]][build-status]
-[![GitHub tag (latest SemVer)][badge-semver]][releases-github]
-[![Terraform Version][badge-terraform]][releases-terraform]
-[![AWS Provider Version][badge-tf-aws]][releases-aws-provider]
-[![Join Slack][badge-slack]][slack]
+[![Build Status](https://github.com/mineiros-io/terraform-aws-cognito-user-pool/workflows/Tests/badge.svg)](https://github.com/mineiros-io/terraform-aws-cognito-user-pool/actions)
```
Comment (kassianh) [inline README.md:3]: Old image URL was [this](https://github.com/mineiros-io/terraform-aws-lambda-function/workflows/CI/CD%20Pipeline/badge.svg). New one (following template layout) doesn't seem to exist. How can I fix this?

**Comment by kassianh** [comment]
Are the badges correct?

File: README.tfdoc.hcl [hunk 2]
```diff
@@ -0,0 +1,1264 @@
...
+
+        variable "domain" {
+          type        = string
+          description = <<-END
+            The domain name that should be used. Can be set to a FQDN or prefix.
+              If no FQDN and `certificate_arn` are set, the domain prefix will be used for
+              the sign-up and sign-in pages that are hosted by Amazon Cognito,
+              e.g. `https://{YOUR_PREFIX}.auth.eu-west-1.amazoncognito.com`.
+              The prefix must be unique across the selected AWS Region.
+              Domain names can only contain lower-case letters, numbers, and hyphens.
```
Comment (mariux) [inline README.tfdoc.hcl:592]: wrong indent.

File: README.tfdoc.hcl [hunk 3]
```diff
@@ -0,0 +1,1264 @@
...
+  section {
+    title   = "Module Outputs"
+    content = <<-END
+      The following attributes are exported by the module:
+
+      - **`user_pool`**: The `cognito_user_pool` object.
+      -**`domain`**: The full `aws_cognito_user_pool` object.
+      - **`clients`**: A map of `cognito_user_pool_client` objects. The map is keyed by the `name` of the created clients. Client secrets are filtered out of this map and are available through the `client_secrets` output variable and flagged as sensitive.
+      - **`client_secrets`**: A sensitive map of client secrets for all created `cognito_user_pool_client` resources. The map is keyed by the `name` of the created clients.
+      -**module_enabled**: Whether the module is enabled.
```
Comment (mariux) [inline README.tfdoc.hcl:1070]: please fix format ;) we really need to support this in terradoc ;)

**Review by mariux — CHANGES_REQUESTED** [review]
minor indent and style issues...

thanks for catching all the other bugs.. ;)

File: README.md (hunk 1 above)
Comment (thiesen) [reply README.md:3]: hey @kassianh, the URL of the badge is formed using the GH workflow name as part of it, like `https://github.com/mineiros-io/terraform-aws-cognito-user-pool/workflows/<WORKFLOW_NAME>/badge.svg`.

in the majority of repos, we have the workflow name set to `Tests` but in this case the name is `CI/CD Pipeline` as you can see in https://github.com/mineiros-io/terraform-aws-cognito-user-pool/actions/workflows/main.yml

Comment (kassianh) [reply README.md]: Thank you! This makes sense.

File: README.tfdoc.hcl [hunk 4]
```diff
@@ -0,0 +1,1261 @@
...
+section {
+  title   = "terraform-aws-cognito-user-pool"
+  toc     = true
+  content = <<-END
+    A [Terraform] module for deploying and managing
+    [Cognito User Pools]
+    on [Amazon Web Services (AWS)][AWS].
+
+    *This module supports Terraform v1.x, v0.15, v0.14, v0.13 as well as v0.12.20 and above
+    and is compatible with the terraform AWS provider v3.19 and above.*
```
Comment (soerenmartius) [inline README.tfdoc.hcl:45]:
```suggestion
    and is compatible with the Terraform AWS provider v3.19 and above.*
```

File: README.tfdoc.hcl [hunk 5]
```diff
@@ -0,0 +1,1261 @@
...
+
+    The last version supporting terraform AWS provider v2.x is v0.4.1.
```
Comment (soerenmartius) [inline README.tfdoc.hcl:47]:
```suggestion
    The last version supporting Terraform AWS provider v2.x is v0.4.1.
```

**Review by soerenmartius — DISMISSED** [review]

**Review by soerenmartius — APPROVED** [review]
LGTM! Good Job @kassianh

**Comment by soerenmartius** [comment]
@kassianh feel free to merge your PR :)

**Review by thiesen — APPROVED** [review]
lgtm

**Pull request merged**